src/envisionSemanticTokensProvider.ts

```diff
@@ -63,33 +63,46 @@ class EnvisionSemanticTokensProvider implements vscode.DocumentSemanticTokensPro
         // Your envisionHighlighter.ts tokenizer needs to return the tokenized lines
         for (let i = 0; i < document.lineCount; i++) {
             const line = document.lineAt(i).text;
+            const isLineEmpty = document.lineAt(i).isEmptyOrWhitespace;
             const lineTokens = tokenizer.tokenize(line, state);
 
-            let prevStart = 0;
-            for(let j = 0; j < lineTokens.tokens.length; j++) {
-                const token = lineTokens.tokens[j];
-
-                let nextStart = j + 1 < lineTokens.tokens.length 
-                    ? lineTokens.tokens[j + 1].startIndex 
-                    : line.length;
-
-                const tokenType = tokenTypes.indexOf(token.scopes);
-
-                // token line number, relative to the previous token
-                tokens.push(i == 0 ? 0 : (j == 0 ? 1 : 0));
-
-                // token start character, relative to the previous token 
-                // (relative to 0 or the previous token's start if they are on the same line)
-                tokens.push(token.startIndex - prevStart); 
-
-                // the length of the token. A token cannot be multiline.
-                tokens.push(nextStart - token.startIndex); 
-
-                // will be looked up in SemanticTokensLegend.tokenTypes.
-                tokens.push(tokenType);
-
-                tokens.push(0);  // No modifiers for now, hence 0
-                prevStart = token.startIndex;
+            if (isLineEmpty) {
```

Comment:
@ArsenijeD This works, but it would be nicer to add an inline comment to outline why this logic is needed in the first place. In particular, do you understand why VS Code appears to be differing in behavior from Monaco?

Reply:
I've just added comment!

To answer your question: not very likely. There shouldn't be any difference, since VSCode relies on Monaco editor. Anyways, additional investigation on that topic could give us an answer.